### PR TITLE
Flag changes and minor fixes for Open XL compile

### PR DESCRIPTION
**Pull request by Deigue**
Some of the remaining compiling issues require some updates to the existing ZOS flags and conditionals in code in order to compile properly using Open XL for z/OS. Tested to ensure that XLC remains compiling properly and passes its tests.

(This is one part of the multiple changes added for supporting Open XL compilation on OMR)

### CODE REVIEW

File: third_party/gtest-1.8.0/include/gtest/internal/gtest-port.h [hunk 1]
```diff
@@ -478,7 +478,7 @@ struct _RTL_CRITICAL_SECTION;
 // detecting whether they are enabled or not.  Therefore, we assume that
 // they are enabled unless the user tells us otherwise.
 #  define GTEST_HAS_EXCEPTIONS 1
-# elif defined(__IBMCPP__) && __EXCEPTIONS
+# elif (defined(__IBMCPP__) || defined(__open_xl__) && defined(__cplusplus)) && __EXCEPTIONS
```
Comment (babsingh) [inline third_party/gtest-1.8.0/include/gtest/internal/gtest-port.h:481]: for easier readability, can brackets be added to highlight operator precedence?

File: cmake/modules/platform/os/zos.cmake [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 list(APPEND OMR_PLATFORM_DEFINITIONS
 	-D_ALL_SOURCE
-	-D_OPEN_THREADS=2
+	-D_OPEN_THREADS=3
```
Comment (babsingh) [inline cmake/modules/platform/os/zos.cmake:24]: why the move from 2 to 3 is needed? can we document the justification and effects of non-trivial changes in the commit message?

Comment (Deigue) [reply cmake/modules/platform/os/zos.cmake:24]: Sure, I will add the details to the commit description as well.
After having encountered some compilation issues related to `pthread` methods, the team working on Open XL pointed me towards https://www.ibm.com/docs/en/zos/2.5.0?topic=files-feature-test-macros, and heavily suggested to change to `OPEN_THREADS=3` in order to enable/expose more pthread functions used.

File: third_party/gtest-1.8.0/include/gtest/internal/gtest-port.h (hunk 1 above)
Comment (Deigue) [reply third_party/gtest-1.8.0/include/gtest/internal/gtest-port.h]: Done

File: fvtest/omrGtestGlue/CMakeLists.txt [hunk 3]
```diff
@@ -66,8 +66,13 @@ endif()
 if(OMR_OS_AIX OR OMR_OS_ZOS)
 	list(APPEND OMR_GTEST_DEFINITIONS
 		-DGTEST_ENV_HAS_STD_TUPLE_
-		-D__IBMCPP_TR1__
 	)
+
+	if(NOT OMR_TOOLCONFIG STREQUAL "openxl")
+		LIST(APPEND OMR_GTEST_DEFINITIONS
```
[the remark this answers is not in the document]
Comment (Deigue) [reply fvtest/omrGtestGlue/CMakeLists.txt]: Made the formatting correction here.

File: third_party/gtest-1.8.0/include/gtest/internal/gtest-port.h (hunk 1 above)
Comment (babsingh) [reply third_party/gtest-1.8.0/include/gtest/internal/gtest-port.h]: please apply/extend to other ifdefs, which were modified in this PR.

Comment (Deigue) [reply third_party/gtest-1.8.0/include/gtest/internal/gtest-port.h]: Added brackets to highlight precedence wherever needed.

**Comment by babsingh** [comment]
jenkins build all

**Comment by babsingh** [comment]
jenkins build all

**Comment by babsingh** [comment]
@Deigue The zOS PR build has failed to build: https://ci.eclipse.org/omr/job/PullRequest-zos_390-64/4398/. Please investigate the errors.

@AdamBrousseau Other PR builds failed due to machine issues. Are these known infra issues?

**Comment by Deigue** [comment]
When going to the full log I am seeing:
```
09:01:59  [ 94%] Building CXX object fvtest/rastest/CMakeFiles/omrsubscribertest.dir/__/__/util/main_function/main_function.cpp.o
09:02:00  [ 94%] Linking CXX executable ../../omrsubscribertest
09:02:00  make[2]: *** No rule to make target 'libomrsig.x', needed by 'omrsigtest'.  Stop.
09:02:00  make[2]: *** Waiting for unfinished jobs....
09:02:00  [ 94%] Building CXX object fvtest/sigtest/CMakeFiles/omrsigtest.dir/__/__/util/main_function/main_function.cpp.o
09:02:01  CMakeFiles/Makefile2:3742: recipe for target 'fvtest/sigtest/CMakeFiles/omrsigtest.dir/all' failed
09:02:01  make[1]: *** [fvtest/sigtest/CMakeFiles/omrsigtest.dir/all] Error 2
09:02:01  make[1]: *** Waiting for unfinished jobs....
```
Hinting towards the problem perhaps being something related to `sigTest.cpp` , since the changes there were only the `OPEN_THREADS` tweak, not sure what is the underlying issue behind the `.x` file generated. I do see its part of `omrsigtest` , so might be able to double check and see whats happen by making/running that standalone.

**Comment by Deigue** [comment]
Need to double check this build with XLC (looping in Open J9 compilation)
as well as trying to compile Open XL with Open J9, to ensure it is reaching as far (as per current progress), since the `zos_rename_exports.cmake` tweak was only throwing errors during compilation of Open J9 with Open XL.

**Comment by Deigue** [comment]
@babsingh So I had to tweak stuff around a bit to actually get everything to work as expected with Open XL (these changes are used when compiling with Open J9 with jdk21 zos)

I ended up having to change it to the following below:
```
if(EXISTS ${SRC_FILE} AND NOT "${SRC_FILE}" STREQUAL "${DEST_FILE}")
	file(RENAME "${SRC_FILE}" "${DEST_FILE}")
endif()

# Work around a bug in CMake where it looks for .x files in the runime dir rather than the archive dir.
if(EXISTS ${DEST_FILE} AND NOT "${ARCHIVE_DIR}" STREQUAL "${RUNTIME_DIR}")
	file(COPY "${DEST_FILE}" DESTINATION "${RUNTIME_DIR}")
endif()
```

Still double-checking the XLC build and will get back to you once it clears up.

**Comment by babsingh** [comment]
jenkins build zos

**Comment by Deigue** [comment]
Doing a full XLC build with only OMR branch pointing to this PR and saw this error, not sure if its related, since I can't see any direct relation yet.
```
[ 69%] Building CXX object runtime/compiler/CMakeFiles/j9jit.dir/codegen/J9CodeGenerator.cpp.o
"/jit/team/gauravc/repos/openj9-openjdk-jdk21-zos/openj9/runtime/compiler/runtime/J9CodeCacheManager.hpp", line 59.70: CCN5274 (S) The name lookup for "TR_EnableCodeCacheDisclaiming" did not find a declaration.
CCN0793(I) Compilation failed for file /jit/team/gauravc/repos/openj9-openjdk-jdk21-zos/openj9/runtime/compiler/codegen/J9CodeGenerator.cpp.  Object file not created.
make[6]: *** [runtime/compiler/CMakeFiles/j9jit.dir/build.make:270: runtime/compiler/CMakeFiles/j9jit.dir/codegen/J9CodeGenerator.cpp.o] Error 12
```

**Comment by babsingh** [comment]
jenkins build all

**Review by babsingh — APPROVED** [review]

**Pull request merged**